services/src/test/scala/cromwell/pipeline/service/GitLabProjectVersioningTest.scala

```diff
@@ -45,6 +48,48 @@ class GitLabProjectVersioningTest extends AsyncWordSpec with ScalaFutures with M
         }
       }
     }
+
+    "getFile" should {
+      "return file with 200 response" taggedAs Service in {
+        val project = TestProjectUtils.getDummyProject(ProjectId("1"), UserId("test"), "name-1", "repository-1", false)
+        val path = Paths.get("test.md")
+        val version = Some(Version("master"))
+
+        when(
+          mockHttpClient.get(
+            MockitoMatchers.eq(
+              s"${gitLabConfig.url}/projects/${project.projectId.value}/repository/files/${URLEncoder
+                .encode(path.toString, "UTF-8")}/raw"
+            ),
+            MockitoMatchers.eq(Map("ref" -> version.map(el => el.value).getOrElse("master"))),
+            MockitoMatchers.any[Map[String, String]]
+          )(MockitoMatchers.any[ExecutionContext])
+        ).thenReturn(Future.successful(Response(200, "Test File", Map())))
+
+        gitLabProjectVersioning.getFile(project, path, version).map(_ shouldBe (Right("Test File")))
+      }
+
+      "throw new VersioningException with not 200 response" taggedAs Service in {
+        val project = TestProjectUtils.getDummyProject(ProjectId("1"), UserId("test"), "name-1", "repository-1", false)
```

Comment:
Why you use this construction? 
You can use just a dummy project.

Reply:
Ok, I will leave dummyProject empty.